### DISCUSSION TITLE
Provide link to my own fork when displaying upstream repo

### DISCUSSION THREAD

**Discussion by softins** [community/community#9143] [Other Feature Feedback, Questions, & Ideas]
When displaying my own fork of a project, the header includes a useful link to get to the upstream project. For example:
![image](https://user-images.githubusercontent.com/3224952/146800295-5218ccc9-50c1-4a63-946b-4ba6711c18f6.png)

However, I have not found a reciprocal way to easily get from the upstream project to my own fork. All the ways I have found involve multiple steps. I have to either go via "My repos", or display the list of forks and scan down for mine. And the latter takes more steps now that the link to the list of forks has been removed from the heading buttons.

My suggestion is: if displaying the upstream of a project which I have forked, provide a link similar to the one above, to get directly to my own fork. For example (constructed manually):
![image](https://user-images.githubusercontent.com/3224952/146800709-db4856d0-c184-4fba-a0a6-33e06aea90f7.png)

Of course, if I had forked a repo that was itself a fork, it would be necessary to display both links:
"forked from parent/project; forked by me at softins/project"

I hope this is a useful suggestion that could be adopted!

**Comment by idbrii** [comment]
It's not very discoverable, but if you've already forked a project and you click Fork, you'll get a dialog that says "You've already forked {repo}" with a link to your fork.

However, it'd be nice if the fork button looked different to indicate that you've forked this repo.

**Reply by softins** [reply]
That is in itself useful, but not the primary purpose of my suggestion. I don't just want to see if I have already forked the repo being viewed; I probably know that anyway. What I want is a link from the upstream repo page to my own fork. At the moment, it takes several operations to get back to my own fork, compared with one operation to get from my fork to the upstream repo.

**Reply by idbrii** [reply]
Not sure what I described doesn't address the purpose of your request. It gets you from upstream to your fork. However, it requires two clicks and the proposal requires one, so I can see how it might be insufficient.

**Reply by softins** [reply]
It's true that it is a quicker way than any I had found, so will be useful to me - thanks! Although as you said, it's not very discoverable.
I still think my proposal would be much more obvious to people, and does have a certain symmetry and consistency. The information is evidently already available to the page generation, because when I click on Fork, it seems to pop up the "You have already forked" box without a round trip to the server.

What I don't know is how suggestions in this forum might actually reach the people responsible for deciding and implementing. Or indeed who those people are.

**Reply by trasherdk** [reply]
Currently, if unlucky, you have no way to click your way from an upstream to your fork.

Clicking `fork` shows me this:
![image](https://user-images.githubusercontent.com/5003891/163488448-a9ab55b2-a006-44c4-b74f-bb1159c9e561.png)
A message not really relevant to the situation.

Clicking `View existing forks.` shows this:
![image](https://user-images.githubusercontent.com/5003891/163488596-dd59551e-b38d-4154-9184-67ab2a370abc.png)

And my fork is not in the list of forks shown.
